packages/kbn-interpreter/src/common/lib/ast/ast.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export type AstNode = Ast | AstFunction | AstArgument;
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type Ast = {
+  type: 'expression';
+  chain: AstFunction[];
+};
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type AstFunction = {
```

Comment:
i think we should try to get rid of duplicate types for this (ones inside expressions plugin and ones inside kbn-interpreter package. If we add optional debug property on this one i think we can reuse this ones in expressions plugin. not necesarry part of this pr.

Reply:
That's what I tried already. I did that with type augmentation in the expression plugin as the cleanest solution, but that's causing so many type errors with the `PersistableState` generics as they require serializable objects. Fixing those errors caused many changes in other parts of the code using the expressions plugin.

I don't think it would be correct to declare the `debug` property in the interpreter package because it doesn't belong there. It's a custom logic built on top of the tree structure in the expressions plugin.

In the end, I refactored those types in the expressions plugin to extend the related ones from the interpreter. As a result, I removed all the duplicating typings and kept only the ones produced by the expressions plugin. This approach also keeps our public API unchanged.